server/plugin.go

```diff
@@ -452,6 +458,8 @@ func (p *Plugin) syncUsers() {
 
 		username := "msteams_" + slug.Make(msUser.DisplayName)
 		if !isUserPresent {
+			p.storeMutex.Lock()
```

Comment:
Think about - 
Should we lock and unlock for each iteration of the loop OR should we lock once and unlock after the function returns?

Reply:
I think it is fine as it is, as we don't want to lock it if the users are already present